pkg/descheduler/strategies/failedpods.go

```diff
@@ -1,174 +0,0 @@
-		return
-	}
-	// Only list failed pods
-	phaseFilter := func(pod *v1.Pod) bool { return pod.Status.Phase == v1.PodFailed }
-	podFilter = podutil.WrapFilterFuncs(phaseFilter, podFilter)
-
-	for _, node := range nodes {
-		klog.V(1).InfoS("Processing node", "node", klog.KObj(node))
-		pods, err := podutil.ListAllPodsOnANode(node.Name, getPodsAssignedToNode, podFilter)
-		if err != nil {
-			klog.ErrorS(err, "Error listing a nodes failed pods", "node", klog.KObj(node))
-			continue
-		}
-
-		for i, pod := range pods {
-			if err = validateFailedPodShouldEvict(pod, *strategyParams); err != nil {
-				klog.V(4).InfoS(fmt.Sprintf("ignoring pod for eviction due to: %s", err.Error()), "pod", klog.KObj(pod))
-				continue
-			}
-
-			if _, err = podEvictor.EvictPod(ctx, pods[i], node, "FailedPod"); err != nil {
-				klog.ErrorS(err, "Error evicting pod", "pod", klog.KObj(pod))
-				break
-			}
-		}
-	}
-}
-
-func validateAndParseRemoveFailedPodsParams(
-	ctx context.Context,
-	client clientset.Interface,
-	params *api.StrategyParameters,
-) (*validatedFailedPodsStrategyParams, error) {
-	if params == nil {
-		return &validatedFailedPodsStrategyParams{
-			ValidatedStrategyParams: validation.DefaultValidatedStrategyParams(),
-		}, nil
-	}
-
-	strategyParams, err := validation.ValidateAndParseStrategyParams(ctx, client, params)
-	if err != nil {
-		return nil, err
-	}
-
-	var reasons, excludeOwnerKinds sets.String
-	var includingInitContainers bool
-	var minPodLifetimeSeconds *uint
-	if params.FailedPods != nil {
-		reasons = sets.NewString(params.FailedPods.Reasons...)
-		includingInitContainers = params.FailedPods.IncludingInitContainers
-		excludeOwnerKinds = sets.NewString(params.FailedPods.ExcludeOwnerKinds...)
-		minPodLifetimeSeconds = params.FailedPods.MinPodLifetimeSeconds
-	}
-
-	return &validatedFailedPodsStrategyParams{
-		ValidatedStrategyParams: *strategyParams,
-		includingInitContainers: includingInitContainers,
-		reasons:                 reasons,
-		excludeOwnerKinds:       excludeOwnerKinds,
-		minPodLifetimeSeconds:   minPodLifetimeSeconds,
-	}, nil
-}
-
-// validateFailedPodShouldEvict looks at strategy params settings to see if the Pod
-// should be evicted given the params in the PodFailed policy.
-func validateFailedPodShouldEvict(pod *v1.Pod, strategyParams validatedFailedPodsStrategyParams) error {
-	var errs []error
-
-	if strategyParams.minPodLifetimeSeconds != nil {
-		podAgeSeconds := uint(metav1.Now().Sub(pod.GetCreationTimestamp().Local()).Seconds())
-		if podAgeSeconds < *strategyParams.minPodLifetimeSeconds {
-			errs = append(errs, fmt.Errorf("pod does not exceed the min age seconds of %d", *strategyParams.minPodLifetimeSeconds))
-		}
-	}
-
-	if len(strategyParams.excludeOwnerKinds) > 0 {
-		ownerRefList := podutil.OwnerRef(pod)
-		for _, owner := range ownerRefList {
-			if strategyParams.excludeOwnerKinds.Has(owner.Kind) {
-				errs = append(errs, fmt.Errorf("pod's owner kind of %s is excluded", owner.Kind))
```

Comment:
Did we really need all these logs? I moved these conditions to a Filter but happy to add back?

Reply:
They were log level 4 logs. Mostly used only for debugging. Though, they are still a good source of decision making when a user wants to know at which point a pod was ignored.